src/js/views/formButton.f7.jsx

```diff
@@ -0,0 +1,108 @@
+import formItem from "./formItem.f7.jsx";
+
+export default class F7ViewFormButton extends formItem {
+   #AB;
+   #form;
+   constructor(AB, form, definition) {
+      super(definition);
+
+      this.#AB = AB;
+      this.#form = form;
+   }
+
+   async #save() {
+      const AB = this.#AB;
+      const form = this.#form;
+      const definition = this.definition;
+
+      AB.$(`#${definition.id}`).addClass("button-loading");
+
+      const formData = AB.$f7.form.convertToData(`#${form.id}`);
+
+      const parsedFormData = {};
+
+      form.views.forEach((view) => {
+         const field = view.definition.field;
+
+         parsedFormData[field] = view.parseFormData(formData[field]);
+      });
+
+      if (!AB.$f7.input.validateInputs(`#${form.id}`)) {
+         AB.$(`#${definition.id}`).removeClass("button-loading");
+
+         return;
+      }
+
+      const dc = form.datacollection;
+
+      try {
+         if (form.record == null) await dc.model.create(parsedFormData);
+         else
+            await dc.model.update(
+               form.record[dc.datasource.PK()],
+               parsedFormData
+            );
+
+         if (definition.afterSubmitView == null) {
+            AB.$f7.views.current.router.back();
+
+            return;
+         }
+
+         AB.$f7.views.current.router.navigate(definition.afterSubmitView.path, {
```

Comment:
Can we make `definition.kafterSubmitView` be an `.id` for another page?  like we did in the `list.f7.jsx`?

Then we ask for the page and do a `.openView()`

Reply:
Thanks!

Reply:
https://github.com/digi-serve/test_pwa/pull/7/commits/878b6cd792478c3388c5820dbb82b4d9950612ae